src/MongooseHttpServer.cpp

```diff
@@ -268,7 +268,7 @@ void MongooseHttpServer::sendAll(MongooseHttpWebSocketConnection *from, const ch
     if (c == nc) { 
       continue; /* Don't send to the sender. */
     }
-    if (c->flags & MG_F_IS_WEBSOCKET)
+    if (c->flags & MG_F_IS_WEBSOCKET && c->flags & MG_F_IS_MongooseHttpWebSocketConnection)
```

Comment:
This is the critical line. The OCPP socket is stored in the same mg_mgr instance so looping through all connections will get you one OCPP connection eventually. That makes it necessary to check the type of `c->user_connection_data` before casting it to see if it is `MongooseHttpWebSocketConnection` or `MongooseOcppSocketClient`. The Mongoose user flag is just a workaround because RTTI is disabled.

All other typechecks are probably just a matter of style. I thought if a typechecking mechanism is introduced at one place, it should be used at all other places as well (e.g. to prevent others to make errors if they just copy code without a deep understanding of mongoose). If you don't add the other typechecks to the codebase, I'm completely fine with it.